components/widgets/inputs/users_emails_input.jsx

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import AsyncSelect from 'react-select/lib/AsyncCreatable';
+import {components} from 'react-select';
+import {intlShape} from 'react-intl';
+
+import {isEmail} from 'mattermost-redux/utils/helpers';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import MailIcon from 'components/svg/mail_icon';
+import MailPlusIcon from 'components/svg/mail_plus_icon';
+import {imageURLForUser} from 'utils/utils.jsx';
+
+import {t} from 'utils/i18n.jsx';
+
+import './users_emails_input.scss';
+
+export default class UsersEmailsInput extends React.Component {
+    static propTypes = {
+        placeholder: PropTypes.string,
+        usersLoader: PropTypes.func,
+        onChange: PropTypes.func,
+        value: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.object, PropTypes.string])),
+        noOptionsMessageId: PropTypes.string,
+        noOptionsMessageDefault: PropTypes.string,
+        noMatchMessageId: PropTypes.string,
+        noMatchMessageDefault: PropTypes.string,
+        validAddressMessageId: PropTypes.string,
+        validAddressMessageDefault: PropTypes.string,
+        loadingMessageId: PropTypes.string,
+        loadingMessageDefault: PropTypes.string,
+    }
+
+    static contextTypes = {
+        intl: intlShape.isRequired,
+    };
+
+    loadingMessage = () => {
+        if (!this.context.intl) {
+            return 'Loading';
+        }
+
+        return this.context.intl.formatMessage({
+            id: this.props.loadingMessageId || t('widgets.users_emails_input.loading'),
```

Comment:
Could we avoid this by using `defaultProps`. But then I guess we'd have to use a shape rather than two separate pros for id and default?

Reply:
We could use a shape there, we still have the problem to decide how we want to deal with the translations in the components with a generic way.

Reply:
Yeah 0/5 on this for now. We can solve it conclusively next quarter when we do the big campaign on this :)